server/src/main/java/org/opensearch/action/search/SearchQueryAggregationCategorizer.java

```diff
@@ -0,0 +1,172 @@
+    private static final String AVG_AGGREGATION = "avg";
+    private static final String WEIGHTED_AVG_AGGREGATION = "weighted_avg";
+    private static final String MAX_AGGREGATION = "max";
+    private static final String MIN_AGGREGATION = "min";
+    private static final String SUM_AGGREGATION = "sum";
+    private static final String STATS_AGGREGATION = "stats";
+    private static final String EXTENDED_STATS_AGGREGATION = "extended_stats";
+    private static final String FILTER_AGGREGATION = "filter";
+    private static final String FILTERS_AGGREGATION = "filters";
+    private static final String ADJACENCY_MATRIX_AGGREGATION = "adjacency_matrix";
+    private static final String SAMPLER_AGGREGATION = "sampler";
+    private static final String DIVERSIFIED_AGGREGATION = "diversified";
+    private static final String GLOBAL_AGGREGATION = "global";
+    private static final String MISSING_AGGREGATION = "missing";
+    private static final String NESTED_AGGREGATION = "nested";
+    private static final String REVERSE_NESTED_AGGREGATION = "reverse_nested";
+    private static final String GEO_DISTANCE_AGGREGATION = "geo_distance";
+    private static final String HISTOGRAM_AGGREGATION = "histogram";
+    private static final String SIGNIFICANT_TERMS_AGGREGATION = "significant_terms";
+    private static final String SIGNIFICANT_TEXT_AGGREGATION = "significant_text";
+    private static final String DATE_HISTOGRAM_AGGREGATION = "date_histogram";
+    private static final String RANGE_AGGREGATION = "range";
+    private static final String DATE_RANGE_AGGREGATION = "date_range";
+    private static final String IP_RANGE_AGGREGATION = "ip_range";
+    private static final String TERMS_AGGREGATION = "terms";
+    private static final String PERCENTILES_AGGREGATION = "percentiles";
+    private static final String PERCENTILE_RANKS_AGGREGATION = "percentile_ranks";
+    private static final String MEDIAN_ABSOLUTE_DEVIATION_AGGREGATION = "median_absolute_deviation";
+    private static final String CARDINALITY_AGGREGATION = "cardinality";
+    private static final String TOP_HITS_AGGREGATION = "top_hits";
+    private static final String GEO_CENTROID_AGGREGATION = "geo_centroid";
+    private static final String SCRIPTED_METRIC_AGGREGATION = "scripted_metric";
+    private static final String COMPOSITE_AGGREGATION = "composite";
+    private static final String MULTI_TERMS_AGGREGATION = "multi_terms";
+    private static final String OTHER_AGGREGATION = "other";
+    private final SearchQueryCounters searchQueryCounters;
+
+    public SearchQueryAggregationCategorizer(SearchQueryCounters searchQueryCounters) {
+        this.searchQueryCounters = searchQueryCounters;
+    }
+
+    public void incrementSearchQueryAggregationCounters(Collection<AggregationBuilder> aggregatorFactories) {
+        for (AggregationBuilder aggregationBuilder : aggregatorFactories) {
+            if (aggregationBuilder instanceof ValueCountAggregationBuilder) {
```

Comment:
Hmm, I'm not sure if it is a good practice to write so many verbose if else statements. There might be better ways to achieve this.

For example, we can create a static mapping class (or use a singleton) to record the builder to tag mapping, and use a for loop here to have better readability.

Or, should we consider making the tag as an attribute of the aggregation builder itself, and we can create a generic function in the aggregation builder, so that we can directly add the corresponding counter based on tag.